site/tests/app/controllers/course/CourseMaterialsControllerTester.php

```diff
@@ -297,4 +297,31 @@ public function testModifyCourseMaterialsPermission() {
 
         $this->assertEquals($expected_json, $json);
     }
+
+    public function testRequestedPathUpload() {
+         $this->getFunctionMock('app\controllers\course', 'is_uploaded_file')
+            ->expects($this->any())
+            ->willReturn(true);
+
+        $controller = new CourseMaterialsController($this->core);
+        $_FILES = array();
+        $_POST['requested_path'] = 'foo/foo2';
+
+        $name = "foo.txt";
+        file_put_contents($this->upload_path . "/" .  $name, 'a');
+        $this->buildFakeFile($name);
```

Comment:
Can you go through all testcases here and move the `file_get_contents` function call into the `buildFakeFile` function?